### PR TITLE
Move building to filters

### LINKED ISSUE #3587: Move "Building" from Advanced to normal filters (remove filtering by Floor)

### PR DESCRIPTION
**Pull request by mic4ael**
closes #3587

### CODE REVIEW

File: indico/modules/rb_new/client/js/components/SearchBar.jsx [hunk 1]
```diff
@@ -38,157 +37,36 @@ export default class SearchBar extends React.Component {
...
-
-        if (filterName === 'building') {
-            stateChanges.floor = null;
-        }
-
-        this.setState(stateChanges, () => {
-            const inputTextValue = this.composeInputTextValue();
-            this.setState({inputTextValue, filtersChanged: this.didFiltersChange()});
-            setFilterParameter('text', inputTextValue);
+            setFilterParameter('text', filterValue ? filterValue : null);
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/components/SearchBar.jsx]: `filterValue || null`

File: indico/modules/rb_new/client/js/components/SearchBar.jsx [hunk 2]
```diff
@@ -38,157 +37,36 @@ export default class SearchBar extends React.Component {
...
-
-        return roomName !== filters.text || building !== filters.building || floor !== filters.floor;
+        this.setState({inputTextValue: null});
     };
 
     updateTextFilter = (filterValue) => {
         const {actions: {setFilterParameter}} = this.props;
-        const {building, text: roomName, floor} = parseSearchBarText(filterValue);
-        const stateUpdates = {inputTextValue: filterValue, building, floor, roomName};
+        const stateUpdates = {inputTextValue: filterValue};
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/components/SearchBar.jsx]: I'd just pass it directly to `setState`

File: indico/modules/rb_new/client/js/modules/roomList/RoomFilterBar.jsx [hunk 3]
```diff
@@ -58,11 +59,26 @@ const equipmentRenderer = ({equipment}) => {
     }
 };
 
+// eslint-disable-next-line react/prop-types
+const buildingRenderer = ({building}) => {
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/modules/roomList/RoomFilterBar.jsx]: `renderBuilding`?

File: indico/modules/rb_new/client/js/modules/roomList/RoomFilterBar.jsx [hunk 4]
```diff
@@ -109,6 +125,16 @@ class RoomFilterBarBase extends React.Component {
             <Button.Group size="large">
                 <Button icon="filter" as="div" disabled />
                 <FilterBarController>
+                    <FilterDropdownFactory name="building"
+                                           title={<Translate>Building</Translate>}
+                                           form={({building: selectedBuilding}, setParentField) => (
+                                               <BuildingForm setParentField={setParentField}
+                                                             buildings={Object.keys(buildings)}
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/modules/roomList/RoomFilterBar.jsx]: If all we ever use here is the list of building numbers, then let's add a selector for this:

```js
export const getBuildingNames = createSelector(
    getBuildings,
    buildings => Object.keys(buildings),
);
```

Comment (ThiefMaster) [reply indico/modules/rb_new/client/js/modules/roomList/RoomFilterBar.jsx]: Actually, I just realized that we don't use the `getBuildings` selector anywhere else and the floor information isn't needed anymore. So we can just simplify `getBuildings` and remove all the now-unnecessary logic from it:

```js
export const getBuildings = createSelector(
    getAllRooms,
    (allRooms) => Array.from(new Set(allRooms.map(room => room.building)))
);
```

File: indico/modules/rb_new/client/js/components/SearchBar.jsx [hunk 5]
```diff
@@ -38,157 +37,35 @@ export default class SearchBar extends React.Component {
...
-
-        return textParts.length ? textParts.join(' ') : null;
-    };
-
-    didFiltersChange = () => {
-        const {filters} = this.props;
-        const {building, floor, roomName} = this.state;
-
-        return roomName !== filters.text || building !== filters.building || floor !== filters.floor;
+        this.setState({inputTextValue: null});
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/components/SearchBar.jsx]: Do we even need this? State is lost when a component is unmounted.

Comment (mic4ael) [reply indico/modules/rb_new/client/js/components/SearchBar.jsx]: Good catch, we don't

**Pull request merged**